File: 0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def inorderTraversal(self, root):
        """
        :type root: Optional[TreeNode]
        :rtype: List[int]
        """
        if root == None:
            return []
        
        l = []

        self.recurssionSolution(root, l)
        return l


        


    
    def recurssionSolution(self, r, l):

        if r.left:
            self.recurssionSolution(r.left,  l)
        
        l.append(r.val)

        if r.right:
            self.recurssionSolution(r.right,  l)
        
        return
        

File: 0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

import java.util.ArrayList;
import java.util.List;
 
class Solution {
    public List<Integer> inorderTraversal(TreeNode root) {
        List<Integer> numbers = new ArrayList<>();
        if(root == null) return numbers;
        inAction(root, numbers);
        return  numbers;
    }

    public void inAction(TreeNode root, List<Integer> list){
        if(root.left != null){
          inAction(root.left, list);
        }
        list.add(root.val);
        if(root.right != null){
            inAction(root.right, list);
        }
        return;
    }
}